src/proxy/processors/push-action/captureSSHKey.ts

```diff
@@ -0,0 +1,99 @@
+import { Action, Step } from '../../actions';
+import { SSHKeyForwardingService } from '../../../service/SSHKeyForwardingService';
+import { SSHKeyManager } from '../../../security/SSHKeyManager';
+
+/**
+ * Capture SSH key for later use during approval process
+ * This processor stores the user's SSH credentials securely when a push requires approval
+ * @param {any} req The request object
+ * @param {Action} action The push action
+ * @return {Promise<Action>} The modified action
+ */
+const exec = async (req: any, action: Action): Promise<Action> => {
+  const step = new Step('captureSSHKey');
+
+  try {
+    // Only capture SSH keys for SSH protocol pushes that will require approval
+    if (action.protocol !== 'ssh' || !action.sshUser || action.allowPush) {
+      step.log('Skipping SSH key capture - not an SSH push requiring approval');
+      action.addStep(step);
+      return action;
+    }
+
+    // Check if we have the necessary SSH key information
+    if (!action.sshUser.sshKeyInfo) {
+      step.log('No SSH key information available for capture');
+      action.addStep(step);
+      return action;
+    }
+
+    const authContext = req?.authContext ?? {};
+    const sshKeyContext = authContext?.sshKey;
+    const privateKeySource =
```

Comment:
Is is necessary to have multiple different sources for the private key? If necessary, it'd be nice to have this fallback chain documented somewhere so it's easier for users to configure.

Reply:
Check: Which of these are correct - then remove the ones that are no longer relevant